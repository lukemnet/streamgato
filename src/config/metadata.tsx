const metadata = {
  title: "StreamGato",
};

export default metadata;
