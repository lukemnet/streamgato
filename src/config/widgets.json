[
  {
    "name": "Countdown Timer",
    "route": "/countdowntimer",
    "shorthand": "cd"
  },
  {
    "name": "Stream Timer",
    "route": "/streamtimer",
    "shorthand": "st"
  },
  {
    "name": "Social media widget",
    "route": "/socialmediawidget",
    "shorthand": "sm"
  },
  {
    "name": "Last.fm Recent Song",
    "route": "/lastfmrecentsong",
    "shorthand": "lr"
  }
]