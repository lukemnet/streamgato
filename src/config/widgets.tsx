import countdownTimer from './widgets/countdownTimer';
import streamTimer from './widgets/streamTimer';
import socialMediaWidget from './widgets/socialMediaWidget';
import lastfmRecentSong from './widgets/lastfmRecentSong';

export default [
  countdownTimer,
  streamTimer,
  socialMediaWidget,
  lastfmRecentSong,
];