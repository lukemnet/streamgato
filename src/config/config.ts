import metadata from './metadata/metadata.json';
import countdownTimer from './widgets/countdownTimer.json';
import streamTimer from './widgets/streamTimer.json';
import socialMediaWidget from './widgets/socialMediaWidget.json';
import lastfmRecentSong from './widgets/lastfmRecentSong.json';
import { Widget } from 'types';

const widgetOrigin =
  process.env.REACT_APP_WIDGET_ORIGIN
  || window.location.origin;

const widgets = [
  countdownTimer,
  streamTimer,
  socialMediaWidget,
  lastfmRecentSong,
] as Widget[];

const siteMetadata = {
  widgetOrigin,
  ...metadata,
};

export {
  siteMetadata as metadata,
  widgets,
};
