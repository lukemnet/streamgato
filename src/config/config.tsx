import metadata from './metadata';
import widgets from './widgets';

export { 
  metadata,
  widgets,
};
