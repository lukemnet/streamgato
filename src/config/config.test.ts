import * as config from './config';

it('should match snapshot', () => {
  expect(config).toMatchSnapshot();
})
