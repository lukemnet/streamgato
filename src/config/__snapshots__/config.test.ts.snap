// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "metadata": Object {
    "title": "StreamGato",
    "widgetOrigin": "http://localhost:3000",
  },
  "widgets": Array [
    Object {
      "alias": "ct",
      "name": "Countdown Timer",
      "params": Object {
        "background": Object {
          "info": "Hex, RGB or RGBA. 'transparent' for no color",
          "label": "Background color",
          "shorthand": "bg",
          "type": "string",
          "value": "#000000",
        },
        "borderBottomColor": Object {
          "info": "Hex, RGB or RGBA",
          "label": "Border bottom color",
          "min": 0,
          "shorthand": "bbc",
          "type": "string",
          "value": "",
        },
        "borderBottomSize": Object {
          "label": "Border bottom size (px)",
          "min": 0,
          "shorthand": "bbz",
          "type": "number",
          "value": 0,
        },
        "borderBottomStyle": Object {
          "info": "Available options: dotted, dashed, solid, double, groove, ridge, inset, outset, none",
          "label": "Border bottom style",
          "min": 0,
          "shorthand": "bbs",
          "type": "string",
          "value": "none",
        },
        "borderLeftColor": Object {
          "info": "Hex, RGB or RGBA",
          "label": "Border left color",
          "min": 0,
          "shorthand": "blc",
          "type": "string",
          "value": "",
        },
        "borderLeftSize": Object {
          "label": "Border left size (px)",
          "min": 0,
          "shorthand": "blz",
          "type": "number",
          "value": 0,
        },
        "borderLeftStyle": Object {
          "info": "Available options: dotted, dashed, solid, double, groove, ridge, inset, outset, none",
          "label": "Border left style",
          "min": 0,
          "shorthand": "bls",
          "type": "string",
          "value": "",
        },
        "borderRightColor": Object {
          "info": "Hex, RGB or RGBA",
          "label": "Border right color",
          "min": 0,
          "shorthand": "brc",
          "type": "string",
          "value": "",
        },
        "borderRightSize": Object {
          "label": "Border right size (px)",
          "min": 0,
          "shorthand": "brz",
          "type": "number",
          "value": 0,
        },
        "borderRightStyle": Object {
          "info": "Available options: dotted, dashed, solid, double, groove, ridge, inset, outset, none",
          "label": "Border right style",
          "min": 0,
          "shorthand": "brs",
          "type": "string",
          "value": "",
        },
        "borderTopColor": Object {
          "info": "Hex, RGB or RGBA",
          "label": "Border top color",
          "min": 0,
          "shorthand": "btc",
          "type": "string",
          "value": "",
        },
        "borderTopSize": Object {
          "label": "Border top size (px)",
          "min": 0,
          "shorthand": "btz",
          "type": "number",
          "value": 0,
        },
        "borderTopStyle": Object {
          "info": "Available options: dotted, dashed, solid, double, groove, ridge, inset, outset, none",
          "label": "Border top style",
          "min": 0,
          "shorthand": "bts",
          "type": "string",
          "value": "none",
        },
        "color": Object {
          "info": "Hex, RGB or RGBA",
          "label": "Text color",
          "shorthand": "col",
          "type": "string",
          "value": "#ffffff",
        },
        "customCSS": Object {
          "info": "HTTPS links only",
          "label": "Custom stylesheet URL",
          "shorthand": "cs",
          "type": "string",
          "value": "",
        },
        "fontFamily": Object {
          "label": "Font family",
          "shorthand": "ff",
          "type": "string",
          "value": "Arial",
        },
        "fontSize": Object {
          "label": "Font size (px)",
          "min": 1,
          "shorthand": "fz",
          "type": "number",
          "value": 24,
        },
        "fontStyle": Object {
          "info": "Available options: normal, italic",
          "label": "Font style",
          "shorthand": "fs",
          "type": "string",
          "value": "normal",
        },
        "fontWeight": Object {
          "label": "Font weight",
          "shorthand": "fw",
          "type": "string",
          "value": "bold",
        },
        "horizontalPosition": Object {
          "info": "Available options: left, center, right",
          "label": "Horizontal position (x)",
          "shorthand": "px",
          "type": "string",
          "value": "center",
        },
        "hours": Object {
          "label": "Hours",
          "min": 0,
          "shorthand": "h",
          "type": "number",
          "value": 0,
        },
        "letterSpacing": Object {
          "info": "0 = normal",
          "label": "Letter spacing (px)",
          "shorthand": "ls",
          "type": "number",
          "value": 0,
        },
        "minutes": Object {
          "info": "0 to 59",
          "label": "Minutes",
          "max": 59,
          "min": 0,
          "shorthand": "m",
          "type": "number",
          "value": 0,
        },
        "paddingBottom": Object {
          "label": "Bottom padding (px)",
          "min": 0,
          "shorthand": "pb",
          "type": "number",
          "value": 0,
        },
        "paddingLeft": Object {
          "label": "Left padding (px)",
          "min": 0,
          "shorthand": "pl",
          "type": "number",
          "value": 0,
        },
        "paddingRight": Object {
          "label": "Right padding (px)",
          "min": 0,
          "shorthand": "pr",
          "type": "number",
          "value": 0,
        },
        "paddingTop": Object {
          "label": "Top padding (px)",
          "min": 0,
          "shorthand": "pt",
          "type": "number",
          "value": 0,
        },
        "radiusBottom": Object {
          "label": "Corner radius top-right (px)",
          "min": 0,
          "shorthand": "rtr",
          "type": "number",
          "value": 0,
        },
        "radiusLeft": Object {
          "label": "Corner radius bottom-left (px)",
          "min": 0,
          "shorthand": "rbl",
          "type": "number",
          "value": 0,
        },
        "radiusRight": Object {
          "label": "Corner radius bottom-right (px)",
          "min": 0,
          "shorthand": "rbr",
          "type": "number",
          "value": 0,
        },
        "radiusTop": Object {
          "label": "Corner radius top-left (px)",
          "min": 0,
          "shorthand": "rtl",
          "type": "number",
          "value": 0,
        },
        "seconds": Object {
          "info": "0 to 59",
          "label": "Seconds",
          "max": 59,
          "min": 0,
          "shorthand": "s",
          "type": "number",
          "value": 0,
        },
        "textAlign": Object {
          "info": "Available options: left, center, right",
          "label": "Text align",
          "shorthand": "ta",
          "type": "string",
          "value": "center",
        },
        "verticalPosition": Object {
          "info": "Available options: top, center, bottom",
          "label": "Vertical position (y)",
          "shorthand": "py",
          "type": "string",
          "value": "center",
        },
        "width": Object {
          "info": "0 = width calculated automatically",
          "label": "Container width",
          "shorthand": "w",
          "type": "number",
          "value": 0,
        },
      },
      "route": "/countdowntimer",
    },
    Object {
      "alias": "st",
      "name": "Stream Timer",
      "params": Object {
        "background": Object {
          "label": "Background color",
          "shorthand": "bg",
          "type": "string",
          "value": "#000000",
        },
        "color": Object {
          "label": "Text color",
          "shorthand": "col",
          "type": "string",
          "value": "#ffffff",
        },
        "customCSS": Object {
          "label": "Custom stylesheet URL",
          "shorthand": "cs",
          "type": "string",
          "value": "http://example.org",
        },
        "fontFamily": Object {
          "label": "Font family",
          "shorthand": "ff",
          "type": "string",
          "value": "Arial",
        },
        "fontSize": Object {
          "label": "Font size",
          "shorthand": "fz",
          "type": "number",
          "value": 24,
        },
        "fontStyle": Object {
          "label": "Font style",
          "shorthand": "fs",
          "type": "string",
          "value": "normal",
        },
        "fontWeight": Object {
          "label": "Font weight",
          "shorthand": "fw",
          "type": "string",
          "value": "bold",
        },
        "hours": Object {
          "label": "Hours",
          "shorthand": "h",
          "type": "number",
          "value": 0,
        },
        "letterSpacing": Object {
          "label": "Letter spacing",
          "shorthand": "ls",
          "type": "string",
          "value": "auto",
        },
        "minutes": Object {
          "label": "Minutes",
          "shorthand": "m",
          "type": "number",
          "value": 0,
        },
        "paddingBottom": Object {
          "label": "Bottom padding",
          "shorthand": "pb",
          "type": "number",
          "value": 5,
        },
        "paddingLeft": Object {
          "label": "Left padding",
          "shorthand": "pl",
          "type": "number",
          "value": 5,
        },
        "paddingRight": Object {
          "label": "Right padding",
          "shorthand": "pr",
          "type": "number",
          "value": 5,
        },
        "paddingTop": Object {
          "label": "Top padding",
          "shorthand": "pt",
          "type": "number",
          "value": 5,
        },
        "seconds": Object {
          "label": "Seconds",
          "shorthand": "s",
          "type": "number",
          "value": 0,
        },
        "width": Object {
          "label": "Container width",
          "shorthand": "w",
          "type": "number",
          "value": "200",
        },
      },
      "route": "/streamtimer",
    },
    Object {
      "alias": "sm",
      "name": "Social Media Widget",
      "params": Object {
        "background": Object {
          "label": "Background color",
          "shorthand": "bg",
          "type": "string",
          "value": "#000000",
        },
        "color": Object {
          "label": "Text color",
          "shorthand": "col",
          "type": "string",
          "value": "#ffffff",
        },
        "customCSS": Object {
          "label": "Custom stylesheet URL",
          "shorthand": "cs",
          "type": "string",
          "value": "http://example.org",
        },
        "fontFamily": Object {
          "label": "Font family",
          "shorthand": "ff",
          "type": "string",
          "value": "Arial",
        },
        "fontSize": Object {
          "label": "Font size",
          "shorthand": "fz",
          "type": "number",
          "value": 24,
        },
        "fontStyle": Object {
          "label": "Font style",
          "shorthand": "fs",
          "type": "string",
          "value": "normal",
        },
        "fontWeight": Object {
          "label": "Font weight",
          "shorthand": "fw",
          "type": "string",
          "value": "bold",
        },
        "hours": Object {
          "label": "Hours",
          "shorthand": "h",
          "type": "number",
          "value": 1,
        },
        "letterSpacing": Object {
          "label": "Letter spacing",
          "shorthand": "ls",
          "type": "string",
          "value": "auto",
        },
        "minutes": Object {
          "label": "Minutes",
          "shorthand": "m",
          "type": "number",
          "value": 5,
        },
        "paddingBottom": Object {
          "label": "Bottom padding",
          "shorthand": "pb",
          "type": "number",
          "value": 5,
        },
        "paddingLeft": Object {
          "label": "Left padding",
          "shorthand": "pl",
          "type": "number",
          "value": 5,
        },
        "paddingRight": Object {
          "label": "Right padding",
          "shorthand": "pr",
          "type": "number",
          "value": 5,
        },
        "paddingTop": Object {
          "label": "Top padding",
          "shorthand": "pt",
          "type": "number",
          "value": 5,
        },
        "seconds": Object {
          "label": "Seconds",
          "shorthand": "s",
          "type": "number",
          "value": 30,
        },
        "width": Object {
          "label": "Container width",
          "shorthand": "w",
          "type": "number",
          "value": "200",
        },
      },
      "route": "/socialmediawidget",
    },
    Object {
      "alias": "rs",
      "name": "Last.fm Recent Song",
      "params": Object {
        "apiKey": Object {
          "label": "API key",
          "shorthand": "ak",
          "type": "string",
          "value": "",
        },
        "background": Object {
          "label": "Background color",
          "shorthand": "bg",
          "type": "string",
          "value": "#000000",
        },
        "color": Object {
          "label": "Text color",
          "shorthand": "col",
          "type": "string",
          "value": "#ffffff",
        },
        "customCSS": Object {
          "label": "Custom stylesheet URL",
          "shorthand": "cs",
          "type": "string",
          "value": "http://example.org",
        },
        "fontFamily": Object {
          "label": "Font family",
          "shorthand": "ff",
          "type": "string",
          "value": "Arial",
        },
        "fontSize": Object {
          "label": "Font size",
          "shorthand": "fz",
          "type": "number",
          "value": 24,
        },
        "fontStyle": Object {
          "label": "Font style",
          "shorthand": "fs",
          "type": "string",
          "value": "normal",
        },
        "fontWeight": Object {
          "label": "Font weight",
          "shorthand": "fw",
          "type": "string",
          "value": "bold",
        },
        "letterSpacing": Object {
          "label": "Letter spacing",
          "shorthand": "ls",
          "type": "string",
          "value": "auto",
        },
        "paddingBottom": Object {
          "label": "Bottom padding",
          "shorthand": "pb",
          "type": "number",
          "value": 5,
        },
        "paddingLeft": Object {
          "label": "Left padding",
          "shorthand": "pl",
          "type": "number",
          "value": 5,
        },
        "paddingRight": Object {
          "label": "Right padding",
          "shorthand": "pr",
          "type": "number",
          "value": 5,
        },
        "paddingTop": Object {
          "label": "Top padding",
          "shorthand": "pt",
          "type": "number",
          "value": 5,
        },
        "width": Object {
          "label": "Container width",
          "shorthand": "w",
          "type": "number",
          "value": "200",
        },
      },
      "route": "/lastfmrecentsong",
    },
  ],
}
`;
